signer/src/transaction_signer.rs

```diff
@@ -425,6 +429,14 @@ where
     ) -> Result<(), Error> {
         let db = self.context.get_storage_mut();
 
+        if chain_tip.block_height == self.last_presign_height {
+            tracing::debug!(
+                ?chain_tip,
+                "already processed presign request for this stacks chain tip"
+            );
+            return Ok(());
+        }
+
```

Comment:
Maaaybe we want this to be `<` instead of `==` and maaaybe it is better to store hash here instead of height in case of reorgs